src/random.cpp

```diff
@@ -714,3 +715,9 @@ void RandomInit()
 
     ReportHardwareRand();
 }
+
+std::chrono::microseconds GetExponentialRand(std::chrono::microseconds now, std::chrono::seconds average_interval)
```

Comment:
While here why not
```suggestion
std::chrono::microseconds GetExponentialRand(std::chrono::microseconds now, std::chrono::microseconds average_interval)
```
and drop the following `std::chrono::duration_cast`?

Reply:
I don't think that it is possible: The duration cast is there not because of the type of `average_interval`, but because we do floating-point arithmetic inside of `GetExponentialRand()`:
The expression `unscaled * average_interval + 0.5us` is of type `std::chrono::duration<long double, std::micro>`, no matter whether `average_interval` is of type `std::chrono::seconds` or `std::chrono::microseconds`.

So in order to convert to the integer-type `std::chrono::microseconds`, a `std::chrono::duration_cast` is  necessary anyway.

Reply:
You're right. Sorry.